src/main/scala/mesosphere/marathon/core/instance/update/InstanceUpdateOpResolver.scala

```diff
@@ -43,6 +43,15 @@ private[marathon] class InstanceUpdateOpResolver(
       case op: ReservationTimeout =>
         updateExistingInstance(op.instanceId)(updater.reservationTimeout(_, clock.now()))
 
+      case op: GoalChange =>
+        updateExistingInstance(op.instanceId)(i => {
+          val updatedInstance = i.copy(state = i.state.copy(goal = op.goal))
+          val events = InstanceChangedEventsGenerator.events(updatedInstance, task = None, clock.now(), previousCondition = Some(i.state.condition))
```

Comment:
why? or what would you do instead?

Reply:
Not publish any events. Are we using them anywhere?

Reply:
no... but right now it actually does not publish anything, the events generator does not contain code that would generate events for goal changes... But it can be added later. I guess I can explicitly pass empty events here...

Reply:
We'll definitely need that `InstanceStateChanged` event. Why not add it now?

Reply:
well this is not instance state, it is Instance goal. We had a discussion about this and:
- the current InstanceChange event is very tight to the condition which does not change when goal changes
- since nobody is using these events now we decided not to introduce them at this point
- I am not that sure it makes sense to publish `InstanceGoalChanged` through API - I think that outside users should be OK with condition changes

To sum up - it's not needed right now and it's not obvious how will that event look like and who will consume it :)

Reply:
To add to this, I think this should eventually differentiate between events that inform about a goal change (aka _change of plans_) and events that inform about a conditional change (aka _something happened_). I agree that the former is not used/needed now. 

I'm not sure we should use the `InstanceChangedEventsGenerator` now if it doesn't produce events – the way it is built, it has to infer _what_ has changed, but it only has the previous _condition_. With the current signature, the generator would be completely unable to create an event that informs about a goal change, so why use it?